### PR TITLE
Modify ddmark ExclusiveFields to allow for specifying a one-to-many relationship

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [x] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- In order to specify that neither `NodeFailure` nor `ContainerFailure` are appropriate to set alongside other disruption kinds, I wanted to use ddmark's `ExclusiveField` feature. However, because this has previously only been used to handle pairs of fields, assumed a many-to-many relationship: if any specified field is set, none of the others can. I needed a one-to-many relationship, if the first field is set, the rest cannot, but the rest don't necessarily have an interdependent relationship. This change enables that, without breaking any of the existing uses (at least ours)

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [x] I leveraged continuous integration testing
    - [x] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: ddmark/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ It also allows us to define custom rules to apply to our structures, and focus t
 - ExclusiveFields:
   - `// +ddmark:validation:ExclusiveFields={<fieldName1>,<fieldName2>,...}`
   - Applies to: `<struct>` type
-  - Asserts only one of the given fieldnames isn’t `nil`
+  - Asserts that if `<fieldname1>` can only be non-nil iff all of the other fields are `nil`
```
Comment (Devatoria) [inline ddmark/README.md:33]:
```suggestion
  - Asserts that if `<fieldname1>` can only be non-nil if all of the other fields are `nil`
```

**Review by Devatoria — COMMENTED** [review]
Can we add some extra unit tests to validate the new behavior of exclusive fields?

File: ddmark/README.md (hunk 1 above)
Comment (ptnapoleon) [reply ddmark/README.md:33]: Intentional! Meant "if and only if" here. Is that shorthand unfamiliar? I can remove it then

Comment (nathantournant) [reply ddmark/README.md:33]: Sugg: - Asserts that `<fieldname1>` can only be non-nil if all of the other fields are `nil`

Alternative (seems more readable to me): - Asserts that `<fieldname1>` can be non-nil only if all of the other fields are `nil`
(that's one too many ifs?

Comment (Devatoria) [reply ddmark/README.md:33]: TIL `iff` :D sorry

**Review by Devatoria — APPROVED** [review]

**Pull request merged**